test/test.py

```diff
@@ -99,6 +102,61 @@ def test(dirname=None, config_path=None, force=False):
                 config_path,
                 '\n'.join(diff_lines)))
 
+def test_delay():
+    """
+    Test that r2e waits before fetching repeatedly from the same
+    server.
+    """
+    delay_cfg = """[DEFAULT]
+    to = example@example.com
+    same-server-fetch-interval = 1
+    """
+    tmpdir = tempfile.mkdtemp()
+    cfg_path = _os.path.join(tmpdir, "rss2email.cfg")
+    data_path = _os.path.join(tmpdir, "rss2email.json")
+
+    def r2e(*args):
+        subprocess.call(["r2e", "-c", cfg_path, "-d", data_path] + list(args))
+
+    with open(cfg_path, "w") as f:
+        f.write(delay_cfg)
+
+    num_requests = 3
+
+    def webserver(queue):
+        import http.server, time
+
+        class NoLogHandler(http.server.SimpleHTTPRequestHandler):
+            def log_message(self, format, *args):
+                return
+
+        httpd = http.server.HTTPServer(('', 8000), NoLogHandler)
+        start = 0
+        for _ in range(num_requests):
+            httpd.handle_request()
+            end = time.time()
+            if end - start < 1:
```

Comment:
A second is a bit long for an automated test, something <300ms should be fine.

Reply:
Good suggestion, I changed the test to wait 0.3s. I also needed to change the `same-server-fetch-interval` to be a float rather than an int to wait less than a second.